### ISSUE TITLE
k3s server fails to start with error iptables not found in path

### ISSUE THREAD

**Issue by kellen-dunham** [k3s-io/k3s#2927]
**Environmental Info:**
K3s Version: 
Docker image v1.20.2+k3s1

Node(s) CPU architecture, OS, and Version: 
Linux k3s-host-1 5.10.12-200.fc33.x86_64 #1 SMP Mon Feb 1 02:40:52 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

Cluster Configuration:
Just trying to get the k3s server started.

**Describe the bug:**
I am testing k3s within a podman container. I'm following the same process as with the docker-compose.yaml located in this repo. Every time I start the server pod it immediately fails saying that iptables isn't found in the $PATH. Sure enough I exec into the pod and try to find iptables and it's not found from within the running container.`

**Steps To Reproduce:**
- Created a podman pod yaml file similar to the docker-compose.yaml file.
- ran podman play kube the-pod-file.yaml

**Expected behavior:**
- iptables is located within the k3s container

**Actual behavior:**
- iptables is not located / installed within the k3s container

**Additional context / logs:**
- This behavior happens with release 19 & 20. When using release 18 the server starts up without issue. Besides the image version, nothing else changes on the pod yaml.

**Comment by brandond** [comment]
Can you provide an example pod yaml?

**Comment by kellen-dunham** [comment]
Yes, here is an example config that I'm working with. I added a few comments to indicate what my intent is for the deployment, but overall I'm just trying to get the k3s server to start up and see that the master node is in Ready state.

``` yaml
apiVersion: v1
kind: Deployment
metadata:
  name: k3s-server-1
spec:
  replicas: 1
  selector:
    matchLabels:
      app: k3s-server-1
  template:
    metadata:
      labels:
        app: k3s-server-1
    spec:
      hostname: k3s-server-1
      containers:
      - name: k3s-server-1
        #image: rancher/k3s:v1.20.2-k3s1 (fails: no iptables)
        #image: rancher/k3s:v1.19.7-k3s1 (fails: no iptables)
        image: rancher/k3s:v1.18.15-k3s1 #(OK: has iptables)
        command:
        - server
        - --cluster-cidr
        - 10.60.0.0/16
        - --disable
        - traefik
        - --disable
        - coredns
        - --disable
        - servicelb
        - --disable
        - local-storage
        - --disable
        - metrics-server
        #- --flannel-backend
        #- none
        env:
        - name: K3S_KUBECONFIG_OUTPUT
          value: /output/kubeconfig.yaml
        - name: K3S_KUBECONFIG_MODE
          value: '666'
        securityContext:
          privileged: true
        resources:
          requests:
            memory: "256Mi"
            cpu: "500m"
          limits:
            memory: "1024Mi"
            cpu: "1000m"
        volumeMounts:
        - mountPath: /etc/rancher/node
          name: k3s-server-1-etc-rancher-vhp
        - mountPath: /var/lib/rancher/k3s
          name: k3s-server-1-lib-rancher-vhp
        - mountPath: /output
          name: k3s-server-1-output-vhp
        - mountPath: /run
          name: k3s-server-1-tmpfs0
        - mountPath: /var/run
          name: k3s-server-1-tmpfs1
      restartPolicy: Never # Was always but I couldn't tell where the logs were ending on each fail
      volumes:
      - name: k3s-server-1-etc-rancher-vhp
        hostPath:
          path: /etc/rancher/server-1
          type: Directory
      - name: k3s-server-1-lib-rancher-vhp
        hostPath:
          path: /var/lib/rancher/k3s/server-1/_data
          type: Directory
      - name: k3s-server-1-output-vhp
        hostPath:
          path: /var/lib/rancher/k3s/server-1/output
          type: Directory
      - name: k3s-server-1-tmpfs0 # I don't think podman supports emptyDir right now
        hostPath:
          path: /var/lib/rancher/k3s/server-1/tmpfs0
          type: Directory
      - name: k3s-server-1-tmpfs1
        hostPath:
          path: /var/lib/rancher/k3s/server-1/tmpfs1
          type: Directory
```

**Comment by dweomer** [comment]
Does this deployment work correctly in k8s but fail in podman?

**Comment by brandond** [comment]
> Every time I start the server pod it immediately fails saying that iptables isn't found in the $PATH. Sure enough I exec into the pod and try to find iptables and it's not found from within the running container.

Can you share more info on how you're testing that? It looks like it's there to me:
```
brandond@dev01:~$docker run --rm -it --entrypoint /bin/sh rancher/k3s:v1.20.2-k3s1
/ # echo $PATH
/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/bin/aux
/ # which iptables
/bin/aux/iptables
/ # iptables --version
iptables v1.8.3 (legacy)
```

**Comment by Oats87** [comment]
We are now serving `iptables` binaries in the `bin/aux` folder for the purposes of "falling back" to K3s-provided userspace binaries for `iptables` in the event that a host system does not have `iptables` binaries available. As such, when run in a container, the `iptables` binaries from `bin/aux` should be used as they are the only binaries available due to the user-provided userspace.

As @brandond has alluded to above, is it possible that your environment is modifying the `PATH` environment variable past what we have defined using our Dockerfile? I have been unable to reproduce this issue with both a native K8s environment (using the Deployment) and using `podman play pod.yaml` where my `pod.yaml` is like:
```
apiVersion: v1
kind: Pod
metadata:
  name: k3s-server-1
spec:
      hostname: k3s-server-1
      containers:
      - name: k3s-server-1
        image: rancher/k3s:v1.20.2-k3s1
        #image: rancher/k3s:v1.19.7-k3s1 (fails: no iptables)
        #image: rancher/k3s:v1.18.15-k3s1 #(OK: has iptables)
        command:
        - server
        - --cluster-cidr
        - 10.60.0.0/16
        - --disable
        - traefik
        - --disable
        - coredns
        - --disable
        - servicelb
        - --disable
        - local-storage
        - --disable
        - metrics-server
        #- --flannel-backend
        #- none
        env:
        - name: K3S_KUBECONFIG_OUTPUT
          value: /output/kubeconfig.yaml
        - name: K3S_KUBECONFIG_MODE
          value: '666'
        securityContext:
          privileged: true
        resources:
          requests:
            memory: "256Mi"
            cpu: "500m"
          limits:
            memory: "1024Mi"
            cpu: "1000m"
        volumeMounts:
        - mountPath: /etc/rancher/node
          name: k3s-server-1-etc-rancher-vhp
        - mountPath: /var/lib/rancher/k3s
          name: k3s-server-1-lib-rancher-vhp
        - mountPath: /output
          name: k3s-server-1-output-vhp
        - mountPath: /run
          name: k3s-server-1-tmpfs0
        - mountPath: /var/run
          name: k3s-server-1-tmpfs1
      restartPolicy: Never # Was always but I couldn't tell where the logs were ending on each fail
      volumes:
      - name: k3s-server-1-etc-rancher-vhp
        hostPath:
          path: /etc/rancher/server-1
          type: Directory
      - name: k3s-server-1-lib-rancher-vhp
        hostPath:
          path: /var/lib/rancher/k3s/server-1/_data
          type: Directory
      - name: k3s-server-1-output-vhp
        hostPath:
          path: /var/lib/rancher/k3s/server-1/output
          type: Directory
      - name: k3s-server-1-tmpfs0 # I don't think podman supports emptyDir right now
        hostPath:
          path: /var/lib/rancher/k3s/server-1/tmpfs0
          type: Directory
      - name: k3s-server-1-tmpfs1
        hostPath:
          path: /var/lib/rancher/k3s/server-1/tmpfs1
          type: Directory
```

**Comment by kellen-dunham** [comment]
Thanks @brandond and @Oats87! I found the problem, and it's not with k3s, it's with podman. I ran the docker-compose file, inspected the container's path against the podman's container's path and notice that the /bin/aux was not included on the pod side. So I just updated the deployment to specify the path as it is in docker `"PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/bin/aux"` and it worked as expected. I'll have to look into what's going on with default podman mechanics later, but for now I'm good. Thanks again!
